occurrences/schema.py

```diff
@@ -766,12 +767,18 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
             )
 
             if occurrence.p_event.auto_acceptance:
-                custom_message = enrolment.occurrence.p_event.safe_translation_getter(
+                custom_message: Optional[
+                    str
+                ] = enrolment.occurrence.p_event.safe_translation_getter(
                     "auto_acceptance_message", language_code=person.language
                 )
-                enrolment.approve(custom_message=custom_message)
+                enrolment.approve(send_notification=False)
+                notifiable_enrolments.append((enrolment, custom_message))
             enrolments.append(enrolment)
 
+        for enrolment, custom_message in notifiable_enrolments:
+            enrolment.send_approve_notification(custom_message=custom_message)
+
         return EnrolOccurrenceMutation(enrolments=enrolments)
```

Comment:
Does this work now when `auto_acceptance` is `False`? Based on the code it seems that a message gets sent by a signal handler [here](https://github.com/City-of-Helsinki/palvelutarjotin/blob/7608a837b3aec4a6a530fc097e567b45917a2f0f/occurrences/signals.py#L18-L32), which seems to have the same original problem?

Reply:
The `approve` -method sent the notifications earlier: https://github.com/City-of-Helsinki/palvelutarjotin/pull/275/files/1a15b669c9801470782c6dbd19d978c1f13453ef#diff-6122150bb78e461eabd410f773e54d0014f1bbd9ecc32e368f59631bd7373892L631. Now I just splitted that part of the code to a new model-function that can be called separately.

The signal seems to send the messages only in cases of when the auto acceptance is off. The new code in schema, won't have anything in `notifiable_enrolments` when the auto-acceptance is off, so it won't send in those cases and it never has.

Reply:
> the issue is only with enrolments that are for multiple occurrences all at once. In such cases, the auto acceptance cannot be on

If I understood correctly, I think in that sentence the last word should be `off`? 🙂 

But yeah, if that is the case, then I guess it will work. Seems pretty complicated indeed, and I would say that the signal thing should be refactored away to avoid future problems 🙂 Not needed in this PR ofc if/when this is in hurry.

Reply:
yep, edited the comment :)